theseus/geometry/tests/test_se2.py

```diff
@@ -127,43 +128,5 @@ def test_projection():
     rng.manual_seed(0)
     for _ in range(10):  # repeat a few times
         for batch_size in [1, 20, 100]:
-            se2 = th.SE2.rand(batch_size, generator=rng, dtype=torch.float64)
-            point = th.Point2.rand(batch_size, generator=rng, dtype=torch.float64)
-
-            aux_id = torch.arange(batch_size)
-
             # Test SE2.transform_to
-            def transform_to_func(g, p):
-                return th.SE2(data=g).transform_to(p).data
-
-            jac_raw = torch.autograd.functional.jacobian(
-                transform_to_func, (se2.data, point.data)
-            )
-            jac = []
-            _ = se2.transform_to(point, jac)
-
-            # Check dense jacobian matrices
-            actual = [
-                se2.project(jac_raw[0]),
-                point.project(jac_raw[1]),
-            ]
-
-            expected = [
-                torch.zeros([batch_size, 2, batch_size, 3]).double(),
-                torch.zeros([batch_size, 2, batch_size, 2]).double(),
-            ]
-            expected[0][aux_id, :, aux_id, :] = jac[0]
-            expected[1][aux_id, :, aux_id, :] = jac[1]
-
-            assert torch.allclose(actual[0], expected[0])
-            assert torch.allclose(actual[1], expected[1])
-
-            # Check sparse jacobian matrices
-            actual = [
-                se2.project(jac_raw[0][aux_id, :, aux_id, :], is_sparse=True),
-                point.project(jac_raw[1][aux_id, :, aux_id, :], is_sparse=True),
-            ]
-
-            expected = jac
-            assert torch.allclose(actual[0], expected[0])
-            assert torch.allclose(actual[1], expected[1])
+            check_projection(th.SE2, th.Point2, th.SE2.transform_to, batch_size, rng)
```

Comment:
Is it worth it to check with other group functions as well? (e.g., a loop over `inverse`, `compose`, etc.)

Reply:
Yes, we do have Jacobians for all of these.

Reply:
Added `check_projection_for_compose` and `check_projection_for_inverse` [here](https://github.com/facebookresearch/theseus/blob/41c3e90e283065d15804bd2d439d4d8a02ab6227/theseus/geometry/tests/common.py#L103).